llama_stack/providers/remote/inference/vllm/vllm.py

```diff
@@ -536,7 +538,38 @@ async def openai_embeddings(
         dimensions: int | None = None,
         user: str | None = None,
     ) -> OpenAIEmbeddingsResponse:
-        raise NotImplementedError()
+        self._lazy_initialize_client()
+        assert self.client is not None
+        model_obj = await self._get_model(model)
+
```

Comment:
```suggestion
        assert model_obj.model_type == ModelType.embedding
```
can you add the ModelType validation like in `embeddings` above?